tests/pcap/tests/s2n_client_hellos.rs

```diff
@@ -71,18 +71,33 @@ fn ja4_fingerprints() -> Result<()> {
     test_all_client_hellos(|pcap_hello, s2n_hello| {
         let mut fingerprint = builder.build(&s2n_hello)?;
 
-        let s2n_ja4_hash = fingerprint
+        let s2n_hash = fingerprint
             .hash()
             .context("s2n failed to calculate ja4 hash")?
             .to_owned();
-
-        let s2n_ja4_str = fingerprint
+        let s2n_str = fingerprint
             .raw()
             .context("s2n failed to calculate ja4 string")?
             .to_owned();
 
-        assert_eq!(pcap_hello.ja4_hash(), Some(s2n_ja4_hash));
-        assert_eq!(pcap_hello.ja4_string(), Some(s2n_ja4_str));
+        let mut tshark_hash = pcap_hello
+            .ja4_hash()
+            .expect("pcap did not contain ja4 hash");
+        let mut tshark_str = pcap_hello
+            .ja4_string()
+            .expect("pcap did not contain ja4 string");
+
+        // Handle known issues:
+        // tshark currently doesn't handle special alpn characters correctly
+        // TODO: remove this when tshark updates
+        let exceptions = [("+", "b"), ("-", "2"), ("__", "5f")];
```

Comment:
It's just the new logic I added: https://github.com/aws/s2n-tls/blob/8100442cdeb70d23ab47afe52799f292b7b47956/tls/s2n_fingerprint_ja4.c#L250-L256

You take the first or last character of the hex representation. So "-+" -> "0x2d2b" -> "2b". Although as I type that out I realize I should probably make the conversion always take two characters for clarity, since order matters. I'll update that.

Reply:
Got it. Thank you for clarifying!